contracts/PointTokenVault.sol

```diff
@@ -115,20 +117,22 @@ contract PointTokenVault is UUPSUpgradeable, AccessControlUpgradeable, Multicall
         uint256 cap = caps[address(_token)];
 
         if (cap != type(uint256).max) {
-            if (_amount + _token.balanceOf(address(this)) > cap) {
+            if (totalDeposited[address(_token)] + _amount > cap) {
                 revert DepositExceedsCap();
             }
         }
 
         _token.safeTransferFrom(msg.sender, address(this), _amount);
 
         balances[_receiver][_token] += _amount;
+        totalDeposited[address(_token)] += _amount;
```

Comment:
We should probably set the balance and totalDeposited before the external call to avoid a re-entrancy pattern.

This is just to be extremely conservative. Given we will only ever be giving tokens a cap that we've reviewed an know do not have re-entrancy (but maybe they are upgradeable), and the impact of this re-entrancy here is just evading our cap, meant to limit exposure, not loss of assets and no negative impact to other users.

Reply:
👍  https://github.com/sense-finance/point-tokenization-vault/pull/38/commits/daad4da1583d291422dc287a0b1f6268b2ef0757